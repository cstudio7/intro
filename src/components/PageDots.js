import React, { Component } from 'react';
import {
    StyleSheet,
    Dimensions,
    Animated,
    Image,
    View,
    Text
} from 'react-native';
import PropTypes from 'prop-types';

const PageDots = ({ ...props }) => {

    function numberOfDots() {
        let dots = [];
        for (let i = 0; i < props.numberOfDots; i++) {
            dots.push(i)
        }
        return dots;
    }

    return (
        <View style={styles.container}>
            {numberOfDots().map((dot) =>
                <View
                    key={dot}
                    style={
                        dot === props.currentIndex ?
                            props.lightBackground ?
                                styles.currentDotDark : styles.currentDot
                            :
                            props.lightBackground ?
                                styles.dotDark : styles.dot

                    }
                />
            )}
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        padding: 10,
        zIndex: 2
    },

    currentDot: {
        backgroundColor: '#FFF',
        borderWidth: 1,
        borderColor: '#FFF',
        width: 10,
        height: 10,
        borderRadius: 5,
        margin: 2
    },

    dot: {
        backgroundColor: 'transparent',
        borderWidth: 1,
        borderColor: '#FFF',
        width: 10,
        height: 10,
        borderRadius: 5,
        margin: 2
    },

    currentDotDark: {
        backgroundColor: '#000',
        borderWidth: 1,
        borderColor: '#000',
        width: 10,
        height: 10,
        borderRadius: 5,
        margin: 2
    },

    dotDark: {
        backgroundColor: 'transparent',
        borderWidth: 1,
        borderColor: '#000',
        width: 10,
        height: 10,
        borderRadius: 5,
        margin: 2
    }
});

PageDots.propTypes = {
    numberOfDots: PropTypes.number.isRequired,
    currentIndex: PropTypes.number.isRequired,
    lightBackground: PropTypes.bool.isRequired
}

export default PageDots;
