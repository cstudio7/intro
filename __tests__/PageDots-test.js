import React from 'react';
import renderer from 'react-test-renderer';
import PageDots from '../src/components/PageDots.js';

test('PageDots renders correctly', () => {
    const tree = renderer.create(<PageDots />).toJSON();
    expect(tree).toMatchSnapshot()
});
