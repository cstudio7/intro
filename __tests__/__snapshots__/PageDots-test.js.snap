// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageDots renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 10,
      "zIndex": 2,
    }
  }
/>
`;
